test/system/200-pod.bats

```diff
@@ -237,6 +239,9 @@ EOF
     # confirm that entrypoint is what we set
     run_podman container inspect --format '{{.Config.Entrypoint}}' $infra_cid
     is "$output" "$infra_command" "infra-command took effect"
+    # confirm that infra container name is set
+    run_podman container inspect --format '{{.Name}}' $infra_cid
+    is "$output" "$infra_name" "infra-name took effect"
```

Comment:
Could you verify an error if I set an infra_name to an existing container/pod and make sure it returns a nice error.

Reply:
I'll fix this

Reply:
The current error is: 
```
jvanz@earth:~/repositories/podman> ./bin/podman  pod create --infra-name testing-infra-name 
ERRO[0000] Error freeing pod lock after failed creation: no such file or directory 
Error: error adding Infra Container: error creating container storage: the container name "testing-infra-name" is already in use by "0b25cc744fda353093970052ee80f5dd5c43917590854615ad59dcf5f57f7984". You have to remove that container to be able to reuse that name.: that name is already in use
```

It seems reasonable to me. What do you think? 
I will add a test to validate the expected behaviour.

Reply:
Test added.